### DISCUSSION TITLE
Add AstraDB Support

### DISCUSSION THREAD

**Discussion by waleedzeb** [JanusGraph/janusgraph#2735] [Ideas]
AstraDB is a fully managed serverless Cassandra database. The service manages everything and provides a cost-effective serverless cost structure.

Has anyone tried this?

**Comment by vtslab** [comment]
Thanks for bringing this up. If you want to discuss this further, you can do so in the discussions section or on the [users list](https://lists.lfaidata.foundation/g/janusgraph-users/topics).

AstraDB states that it works with Cassandra 3.11 clients ([https://docs.datastax.com/en/astra/docs/datastax-astra-faq.html](https://docs.datastax.com/en/astra/docs/datastax-astra-faq.html)). JanusGraph supports Cassandra 3.11 ([https://docs.janusgraph.org/changelog/#version-compatibility](https://docs.janusgraph.org/changelog/#version-compatibility)).
Although the [JanusGraph website ](https://janusgraph.org/) mentions ScyllaDB as a separate storage backend, there are many options for managed Cassandra services, so a list of these would not make things clearer and would be out of date soon.

**Reply by porunov** [reply]
Converted the Issue to Discussion

**Comment by vmeln** [comment]
I managed to connect to AstraDB, but the performance was really poor for the writes. I did nothing new that was [described on this thread](https://community.datastax.com/questions/8775/how-do-you-configure-janusgraph-to-connect-to-data.html) to connect.

**Reply by ErickRamirezAU** [reply]
Sorry to hear about your experience. FWIW, vanilla Astra DBs are rate-limited by default to 4K ops/s mostly to avoid surprises on usage (charged at reads/writes per million). We'd be happy to scale you up to millions of ops/s -- you just need to log a request to do that. Cheers! 🍻
